### ISSUE TITLE
Increase Upload Limit for CFA and PA Dataverse

### ISSUE THREAD

**Issue by djbrooke** [IQSS/dataverse.harvard.edu#87]
While we figure out the larger storage strategy for Harvard Dataverse, I'd like to increase upload limits for two dataverses to 50 GB:

https://dataverse.harvard.edu/dataverse/pan
https://dataverse.harvard.edu/dataverse/cfa

Excited to get a few dataverses uploading larger files and seeing how it works!

**Comment by landreev** [comment]
Are we giving them the "direct upload" option too - or just increasing the limit?

**Comment by landreev** [comment]
There's no "code" to review. But below are the command that were used to create and configure the 2 extra stores, "s3cfa" and "s3pan": 

```
/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3cfa.type=s3"
/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3cfa.label=s3cfa"
/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3cfa.bucket-name=dvn-cloud"
/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3cfa.download-redirect=true"
/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3cfa.connection-pool-size=512"
/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3cfa.MaxFileUploadSizeInBytes=50000000000"
/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3cfa.upload-redirect=true"

/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3pan.type=s3"
/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3pan.label=s3pan"
/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3pan.bucket-name=dvn-cloud"
/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3pan.download-redirect=true"
/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3pan.connection-pool-size=512"
/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3pan.MaxFileUploadSizeInBytes=50000000000"
/opt/dvn/app/payara5/bin/asadmin create-jvm-options "\-Ddataverse.files.s3pan.upload-redirect=true"
```

**Comment by landreev** [comment]
(activating these new stores for the dataverses requires a restart - will do in the late pm).

**Comment by qqmyers** [comment]
@landreev - MaxFileUploadSizeInBytes is a setting that can take json to set different limits for different stores (not a jvm option as above)

Also FWIW - I don't think it hurts to make different stores, but if they are going to the same bucket and have the same upload limit, you could just use 1. (Or consider making ~generic directupload stores associated with a size (s3medium, s3large, s3extralarge) rather than specific dataset/dataverse.) Just thinking you may end up with fewer stores overall to manage.

**Comment by landreev** [comment]
@qqmyers - thanks. The second part I knew - just opened these as different stores under the assumption that they may end up being handled separately (maybe even via different buckets). 
But the other thing I was completely missing - thank you...

**Comment by landreev** [comment]
@qqmyers Please note however that I got that option from the 4.20 release notes: 
`The JVM option dataverse.files.<id>.MaxFileUploadSizeInBytes controls the maximum size of file uploads allowed for the given file store.`

**Comment by qqmyers** [comment]
@landreev - good catch - there are two other places where it is correctly referred to as a setting in the release notes for 4.20 and it's correct in the guides.

**Comment by landreev** [comment]
Updated the commands above.

**Comment by landreev** [comment]
(I removed the fake MaxFileUploadSizeInBytes JVM option from the 4.20 release notes)

**Comment by kcondon** [comment]
I initially checked this by looking at the dv general info for storage and also on the dataset upload file size. I did not try an actual file upload. 

A user has reported they cannot replace files on a dataset in one of these dv's, RT 295965 and I have confirmed that and also that you cannot upload any files to a new dataset either. No real server log errors except for a delayed s3 thumbnail complaint.
